src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/close/AttemptCloseStep.kt

```diff
@@ -0,0 +1,47 @@
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.step.close
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.elasticapi.suspendUntil
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.ManagedIndexMetaData
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.CloseActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.Step
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.admin.indices.close.CloseIndexRequest
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+
+class AttemptCloseStep(
+    val clusterService: ClusterService,
+    val client: Client,
+    val config: CloseActionConfig,
+    managedIndexMetaData: ManagedIndexMetaData
+) : Step("attempt_close", managedIndexMetaData) {
```

Comment:
Looks like in the last PR, Drew changed the ```name``` to be within a ```companion object```. Let's change this and ```AttemptOpenStep``` to be consistent with that.

Reply:
I did that because I was consuming that name somewhere else, whereas we are only using this name here so don't necessarily need it outside this class. Can make the choice to either do it for all of them or only the ones that need it.

Reply:
right so since we already have `name` as part of `Step`, I think it is redundant to create another companion object to store the name. I think we should extract the `name` as we really needed them.

Reply:
Okay sure, let's extract ```name``` only as needed.